File: plugin/src/main/java/copilotj/awt/window/IjImageJ.java
/**
 * SPDX-FileCopyrightText: Copyright contributors to the CopilotJ project.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package copilotj.awt.window;

import java.awt.Window;

import copilotj.ImagejListener;
import copilotj.awt.WindowIdentifier;

import ij.ImageJ;

public class IjImageJ extends AbstractAwtWindow<ImageJ> {
  public static class Provider implements AwtWindowProvider {
    @Override
    public AwtWindow tryCreate(final WindowIdentifier identifier, final Window window) {
      if (window instanceof ImageJ) {
        return new IjImageJ(identifier, (ImageJ) window);
      }
      return null;
    }
  }

  private static final String TYPE = "ij.ImageJ";

  public IjImageJ(final WindowIdentifier identifier, final ImageJ window) {
    super(identifier, window, TYPE);
  }

  @Override
  public String describe() {
    return "ImageJ: id=" + this.id;
  }

  @Override
  public AwtWindow.Difference compare(final AwtWindow from, final ImagejListener.HistoryResponse history) {
    if (!(from instanceof IjImageJ)) {
      return super.compare(from, history);
    }

    // always return null, because ImageJ windows are not comparable
    return null;
  }
}

File: copilotj/plugin/awt/window/ij_imagej.py
# SPDX-FileCopyrightText: Copyright contributors to the CopilotJ project.
#
# SPDX-License-Identifier: Apache-2.0

from typing import Literal, override

from copilotj.plugin._base import Verbosity
from copilotj.plugin.awt.window.awt_window import AwtWindowBase

__all__ = ["IjImageJ"]


class IjImageJ(AwtWindowBase[Literal["ij.ImageJ"]]):
    @override
    def _node_name(self) -> str | None:
        return "ImageJ"

    @override
    def _describe(self, *, level: int, verbosity: Verbosity) -> list[str]:
        # The main toolbar's individual buttons aren't useful to enumerate, so
        # render just the window header line.
        return [self._yaml_head()]
